packages/connection/src/Manager.php

```diff
@@ -28,6 +29,155 @@ class Manager implements Manager_Interface {
 	 */
 	protected $secret_callable;
 
+	/**
+	 * A copy of the raw POST data for signature verification purposes.
+	 *
+	 * @var String
+	 */
+	protected $raw_post_data;
+
+	/**
+	 * Verification data needs to be stored to properly verify everything.
+	 *
+	 * @var Object
+	 */
+	private $xmlrpc_verification = null;
+
+	/**
+	 * Creates an instance of the connection manager.
+	 */
+	public function __construct() {
+	}
+
+	/**
+	 * Initializes required listeners. This is done separately from the constructors
+	 * because some objects sometimes need to instantiate separate objects of this class.
+	 */
+	public function init() {
+
+		$is_jetpack_xmlrpc_request = $this->setup_xmlrpc_handlers(
+			$_GET,
+			$this->is_active(),
+			$this->verify_xml_rpc_signature()
+		);
+
+		// All the XMLRPC functionality has been moved into setup_xmlrpc_handlers.
+		if (
+			! $is_jetpack_xmlrpc_request
+			&& is_admin()
+			&& isset( $_POST['action'] ) // phpcs:ignore WordPress.Security.NonceVerification
+			&& (
+				'jetpack_upload_file' === $_POST['action']  // phpcs:ignore WordPress.Security.NonceVerification
+				|| 'jetpack_update_file' === $_POST['action']  // phpcs:ignore WordPress.Security.NonceVerification
+			)
+		) {
+			$this->require_jetpack_authentication();
+			$this->add_remote_request_handlers();
+		} else {
```

Comment:
Could we use `return` at the end of the `if()` block and that way prevent from the need to have everything in an `else` block?

Reply:
Done in b5bc77afac0b66a68e75e2fbb22b2dba113c2fbe